ssl/ssl_lib.c

```diff
@@ -2008,6 +2009,74 @@ int ssl_write_internal(SSL *s, const void *buf, size_t num, size_t *written)
     }
 }
 
+ossl_ssize_t SSL_sendfile(SSL *s, int fd, off_t offset, size_t size, int flags)
+{
+    ossl_ssize_t ret;
+
+    if (s->handshake_func == NULL) {
+        SSLerr(SSL_F_SSL_SENDFILE, SSL_R_UNINITIALIZED);
+        return -1;
+    }
+
+    if (s->shutdown & SSL_SENT_SHUTDOWN) {
+        s->rwstate = SSL_NOTHING;
+        SSLerr(SSL_F_SSL_SENDFILE, SSL_R_PROTOCOL_IS_SHUTDOWN);
+        return -1;
+    }
+
+    if (!BIO_get_ktls_send(s->wbio)) {
+        SSLerr(SSL_F_SSL_SENDFILE, SSL_R_UNINITIALIZED);
+        return -1;
+    }
+
+    ossl_statem_check_finish_init(s, 1);
```

Comment:
This is probably not necessary. It's TLSv1.3 specific and related to early data handling, which shouldn't apply to KTLS.

Reply:
Right. I thought of having this function TLS1.3 ready. Since we are planning on submitting patches for TLS1.3 support soon.
Do you prefer I remove this, and re-introduce it later?

Reply:
Probably its better to remove and re-introduce it later. I'd like to understand how the whole TLSv1.3 solution fits together in a single PR.

Reply:
Done